modules/basicfuncs/basic-funcs.c

```diff
@@ -104,7 +104,8 @@ static Plugin basicfuncs_plugins[] =
   /* misc funcs */
   TEMPLATE_FUNCTION_PLUGIN(tf_env, "env"),
   TEMPLATE_FUNCTION_PLUGIN(tf_template, "template"),
-  TEMPLATE_FUNCTION_PLUGIN(tf_urlencode, "urlencode")
+  TEMPLATE_FUNCTION_PLUGIN(tf_urlencode, "urlencode"),
+  TEMPLATE_FUNCTION_PLUGIN(tf_urlencode, "url-encode"),
```

Comment:
Users might got used to that hyphens and underscores are interchangeable. Should we support "url_encode" as well?

Reply:
no, the reason _ and - is interchangeable as it wasn't consistent in the early years. the documented/preferred form is always the dash based one. (btw we could issue a warning if that's not the case).

When the first template functions were introduced, the dash based convention was already present, so I wouldn't want to publish these with underscores as well.

Reply:
Ok I see.